### DISCUSSION TITLE
bash

### DISCUSSION THREAD

**Discussion by jamilesoares** [community/community#127600] [Programming Help] [answered]
### Select Topic Area

Bug

### Body

I'm trying to do the ./credit function, but it's not showing in my terminal window, it's showing bash: ./credit: it's a directory. I can't finish my codes anymore because of this, can anyone help me?
![21455475-a315-437c-a873-85c31e2e01e2](https://github.com/community/community/assets/160682284/4f550535-73b0-4d9a-b7ad-54984d046917)

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @jamilesoares ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by RafaelJohn9** [comment]
sure am happy to help @jamilesoares 
first
> ls
- type that in the bash terminal and paste the screenshot
- second I'll need the name of the file you want to execute

**Comment by Hadikamali** [comment]
Hi @jamilesoares 
The error you're encountering, `bash: ./credit: Is a directory`, indicates that there is a directory named `credit` in your current working directory, and you're trying to execute it as if it were a file.

Here's how you can resolve this issue:

1. **Rename or Remove the Directory:**
   If the `credit` directory is not needed, you can remove it or rename it. Use the following commands to remove the directory:
   ```bash
   rm -r credit
   ```
   If you want to rename it instead:
   ```bash
   mv credit credit_dir
   ```

2. **Create or Ensure the `credit` File Exists:**
   If you are trying to compile a C program called `credit.c`, ensure that the source file exists and then compile it correctly. For example:
   ```bash
   gcc credit.c -o credit
   ```

3. **Run the Compiled Program:**
   After compiling the program, you should be able to run it with:
   ```bash
   ./credit
   ```

Here is a summary of steps to ensure the process is correct:

1. Ensure there is no directory named `credit` in your working directory:
   ```bash
   rm -r credit  # or rename it using mv credit credit_dir
   ```

2. Compile your program:
   ```bash
   gcc credit.c -o credit
   ```

3. Execute your compiled program:
   ```bash
   ./credit
   ```

If you still encounter issues, please provide more details about what the `credit` directory contains and what exactly you are trying to achieve with the `make credit` command.

Powerd by AI

**Comment by jamilesoares** [comment]
Thank u for your help, but it didn't work. I'll try others options.

**Reply by RafaelJohn9** [reply]
@jamilesoares  you just do what I told you above and I can help after that

**Reply by ebndev** [reply]
Hi @RafaelJohn9, thanks for taking the time to answer!

@jamilesoares if the response above helped, please feel encouraged to mark it as the answer. If not, let us know if you have further questions or post the solution if you've resolved it on your own. It would be helpful for other users who might experience this in the future 😄

**Comment by Interstellar2424** [comment]
The error is occurring because you are on home in your terminal to solve this problem :
<hr>
- Type "cd credit" in your case
<br>
- Then type "make credit" in your case
<br>
- And finally type "./credit" in your case again and then your problem will be solved.
<br>
And you have to do this whenever you are running your program.

**Comment by jamilesoares** [comment, marked as answer]
Thank you for everyone's help! I finally managed to solve it, I was almost giving up, I'm glad to have quick answers here.
